### ISSUE TITLE
Publication Form - Investigate auto-generating the movie filename

### ISSUE THREAD

**Issue by adelikat** [adelikat/tasvideos#43]
Right now it is a publisher entered field, could we automate this?
is 1234M.bk2 enough?
Or maybe we could contct author names, gamename, and branch? what about version?

**Comment by darkpsy3934** [comment]
I think the movie ID as the filename would be enough.  Perhaps something like this would be better:

1234M [branch]-[region].fileext

**Comment by adelikat** [comment]
I like this idea

**Comment by bigbass1997** [comment]
Is this something we still want to implement?

---

Since publication details can change without also updating the movie file, I think it might be best to limit the filename to only data that can't become outdated. Besides the movie ID, region is probably the only detail that shouldn't ever change but region doesn't do much to differentiate files.

So I believe that leaves us with `1234M.ext` which sounds good to me.

Also, if we decide to do this, maybe we want something similar for submission movies for consistency?

**Comment by vadosnaprimer** [comment]
Dunno if anyone stores a bunch of movies on the drive, if they do then relying on IDs would be not ideal. Regular users most likely don't, so I'm not sure if actual names are still useful. Movie names are used when we upload encodes to archive.org iirc.

**Comment by fsvgm777** [comment]
I feel like having the filename just be the publication ID is not really useful, On several published BK2s we have, the author is listed as "default user", so you'd have to go to the publication itself to find the relevant info regardless.

Maybe there's a way to automatically generate a filename based on game name, branch, author(s) and TAS version by the same author(s), making adjustments as necessary?

**Comment by bigbass1997** [comment]
> Dunno if anyone stores a bunch of movies on the drive, if they do then relying on IDs would be not ideal.

I do. Several hundred at this point. In my experience, trying to tie a movie file back to which publication it came from using the original filename is rather annoying, if possible at all. Some authors have multiple pubs for the same game, and some games have so many different pubs that it's practically impossible to figure out which movie came from where. Oftentimes the branch and/or game name have also been abbreviated.

Personally, I rename all the movies I download to just the ID and M (pub), S (sub), or U (userfile). But I recognize my use case (console verification) is surely in the minority. (and is largely irrelevant anyways since I've automated most of this)

I figured that short-term use for most people would benefit from more details. But out-of-date details could be confusing. Maybe including the ID with other details is sufficient?

---

> Movie names are used when we upload encodes to archive.org iirc.

They are yes, though I believe commas get stripped out while uploading ([example](https://tasvideos.org/Publications/Edit/2062)).

> Maybe there's a way to automatically generate a filename based on game name, branch, author(s) and TAS version by the same author(s), making adjustments as necessary?

Yeah I had that idea too. Looking more into it, I see that the primary movie file is denoted as such. So I suspect renaming the file whenever a change is made is doable.

One concern I have though is that the author list can become incredibly long. Most modern filesystems can't handle filenames over 255 characters or bytes. I believe our longest so far is an SM64 movie at 143 chars. With the potential for even longer from communities like Celeste, we'll want to decide how to limit this.

**Comment by fsvgm777** [comment]
> One concern I have though is that the author list can become incredibly long. Most modern filesystems can't handle filenames over 255 characters or bytes. I believe our longest so far is an SM64 movie at 143 chars. With the potential for even longer from communities like Celeste, we'll want to decide how to limit this.

Good point. Maybe if the filename exceeds 255 characters, the authors list could probably be replaced by just "gamenamecommunity" or just "community". But that'll need to be decided on a policy level.
